include/wx/generic/grideditors.h

```diff
@@ -373,6 +373,48 @@ class WXDLLIMPEXP_ADV wxGridCellAutoWrapStringEditor : public wxGridCellTextEdit
     wxDECLARE_NO_COPY_CLASS(wxGridCellAutoWrapStringEditor);
 };
 
+#if wxUSE_DATEPICKCTRL
+
+class WXDLLIMPEXP_ADV wxGridCellDateEditor : public wxGridCellEditor
+{
+public:
+    explicit wxGridCellDateEditor();
+
+    virtual void Create(wxWindow* parent,
+                        wxWindowID id,
+                        wxEvtHandler* evtHandler) wxOVERRIDE;
+
+    virtual void SetSize(const wxRect& rect) wxOVERRIDE;
+
+    virtual void PaintBackground(wxDC& dc,
+                                 const wxRect& rectCell,
+                                 const wxGridCellAttr& attr) wxOVERRIDE;
+
+    virtual bool IsAcceptedKey(wxKeyEvent& event) wxOVERRIDE;
+    virtual void BeginEdit(int row, int col, wxGrid* grid) wxOVERRIDE;
+    virtual bool EndEdit(int row, int col, const wxGrid* grid,
+                         const wxString& oldval, wxString *newval) wxOVERRIDE;
+    virtual void ApplyEdit(int row, int col, wxGrid* grid) wxOVERRIDE;
+
+    virtual void Reset() wxOVERRIDE;
+
+    virtual wxGridCellEditor *Clone() const wxOVERRIDE;
+
+    virtual wxString GetValue() const wxOVERRIDE;
+
+protected:
+    wxDatePickerCtrl* DatePicker() const;
+
+protected:
+    wxDateTime m_value;
+    wxSize m_bestSize;
```

Comment:
I'm fine with using the best size (instead of the full cell size) for the size of the date picker, my postscript just meant that, AFAICS, we don't need to store it as a member variable, as it's only used in a single member function.